pkg/scheduler/framework/runtime/framework.go

```diff
@@ -354,6 +376,37 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 	return f, nil
 }
 
+func fillEventToPluginMap(p framework.Plugin, eventToPlugins map[framework.ClusterEvent]sets.String) {
+	ext, ok := p.(framework.EnqueueExtensions)
+	if !ok {
+		// If interface EnqueueExtensions is not implemented, register the default events
+		// to the plugin. This is to ensure backward compatibility.
+		registerClusterEvents(p.Name(), eventToPlugins, allClusterEvents)
+		return
+	}
+
+	events := ext.EventsToRegister()
+	// It's rare that a plugin implements EnqueueExtensions but returns nil.
+	// We treat it as: the plugin is not interested in any event, and hence pod failed by that plugin
+	// cannot be moved by any regular cluster event.
+	if len(events) == 0 {
+		klog.Warningf("it's rare have a plugin implemented EnqueueExtensions but returns nil ", "plugin", p.Name())
```

Comment:
use `InfoS`

Reply:
Updated. Just curious: is Waningf deprecated?

Reply:
Yes